lib/hanami/provider/source.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Hanami
+  module Provider
+    class Source < Dry::System::Provider::Source
+      attr_reader :slice
+
+      alias_method :target_container, :slice
+      alias_method :target, :slice
```

Comment:
I don't love that I had to add both of these lines here.

But I think this can be temporary.

If we update `Dry::System::Provider::Source`, and replace `alias_method :target, :target_container` with `def target = target_container`, then I think we can reduce these lines here to just:

```ruby
def target_container = slice
```

Which I think will be a good final state :)

Reply:
Well this appears to have totally blown up the suite so I will need to investigate further

Reply:
@alassek Where have things blown up? This PR looks green to me.

Reply:
@alassek after my extra adjustment to https://github.com/dry-rb/dry-system/pull/275, I've now pushed another commit to this branch with just the single `def target_container = slice` override, and now everything is looking good 👍🏼